include/SPTPersistentCache/SPTPersistentCache.h

```diff
@@ -227,16 +231,25 @@ typedef NSString * _Nonnull(^SPTPersistentCacheChooseKeyCallback)(NSArray<NSStri
 - (void)unscheduleGarbageCollector;
 /**
  * Delete all files files in managed folder unconditionaly.
+ * @param callback May be nil if not interested in result.
+ * @param queue Queue on which to run the callback. If callback is nil this is ignored otherwise mustn't be nil.
  */
-- (void)prune;
+- (void)pruneWithCallback:(SPTPersistentCacheResponseCallback _Nullable)callback
+                  onQueue:(dispatch_queue_t _Nullable)queue;
 /**
  * Wipe only files that locked regardless of refCount value.
+ * @param callback May be nil if not interested in result.
+ * @param queue Queue on which to run the callback. If callback is nil this is ignored otherwise mustn't be nil.
  */
-- (void)wipeLockedFiles;
+- (void)wipeLockedFilesWithCallback:(SPTPersistentCacheResponseCallback _Nullable)callback
+                            onQueue:(dispatch_queue_t _Nullable)queue;
 /**
  * Wipe only files that are not locked regardles of their expiration time.
+ * @param callback May be nil if not interested in result.
+ * @param queue Queue on which to run the callback. If callback is nil this is ignored otherwise mustn't be nil.
  */
-- (void)wipeNonLockedFiles;
+- (void)wipeNonLockedFilesWithCallback:(SPTPersistentCacheResponseCallback _Nullable)callback
```

Comment:
Same here, could we make this an additive change instead, keeping the old method around so we don’t break the current API?